pkg/isb/forward/forward.go

```diff
@@ -318,6 +319,10 @@ func (isdf *InterStepDataForward) writeToBuffers(ctx context.Context, messageToS
 
 // writeToBuffer forwards an array of messages to a single buffer and is a blocking call or until shutdown has been initiated.
 func (isdf *InterStepDataForward) writeToBuffer(ctx context.Context, toBuffer isb.BufferWriter, messages []isb.Message) (writeOffsets []isb.Offset, err error) {
+	var totalBytes float64
+	for _, m := range messages {
```

Comment:
is there a way we can avoid this extra loop of O(N)? Can we collect this info while writing to buffer? We can do summation while querying from prometheus.

Reply:
Based on our implementation, it's going to be very tricky, because we do batch writing and need to consider failure case.